### DISCUSSION TITLE
Strings not updated in Weblate after Github pull

### DISCUSSION THREAD

**Discussion by ywarnier** [WeblateOrg/weblate#10782] [Q&A]
Hi,

First of all, thanks for Weblate, it is impressive in its features list and well designed as far as I could tell so far.
I'm having difficulties finding my way around the extensive documentation at https://docs.weblate.org/ (I feel the installation documentation talks too much about countless options), but I fell back to installing Weblate through Docker and now I have it importing my .po files through a public Github repo, translating strings in new languages and downloading them (manually - couldn't get the Github push to work yet) and push them inside our repository. All good on that front.

When I push a new string in my source .po file (I also do that in the .pot), Weblate sees the new commit, updates the local repository (or says it is processing it), but does not seem to find the new string: it does not appear in the list of strings to be translated.

I'm a bit lost as to how to find information about what's going on.

When I go to the component's "Repository maintenance" tab, I see the last commit shown in "Last remote commit", and after a few hours of the new string not appearing, I launched a rescan ("Rescan all translation files in the Weblate repository"), which seems to be executing (as per `docker-compose logs` showing `weblate_1   | gunicorn stderr | [2024-01-13 18:41:55,651: INFO/446] chamilo-lms/translations: updating completed`), but the string does not appear in my component's translations.

I have a main.pot *and* a messages.en.po which are pretty much the same (I wanted to use main.pot as source and messages.en.po as the "perfected" English translation, but I've read that I should only use messages.en.po), and I see in the logs that I get: `weblate_1   | gunicorn stderr | [2024-01-13 18:41:55,073: WARNING/446] chamilo-lms/translations: duplicate language found: en (en, en)` in the middle of the other language scans items, so I tried reloading the component giving messages.en.po as a "template for new files", but that didn't help either. There are more than 40 .po files in total, English is just one of them, and the one we used as a source.

How can I make sure that new strings added to my repo are correctly added to the strings in Weblate?

**Comment by nijel** [comment]
New strings should appear, so there is most likely some misconfiguration of your translations.

Are these bilingual or monolingual files? See https://docs.weblate.org/en/latest/formats.html#bilingual-and-monolingual-formats. For monolingual files, the strings are determined by [Monolingual base language file](https://docs.weblate.org/en/latest/admin/projects.html#component-template).

**Comment by emmapeel2** [comment]
In my experience, if there are duplicated languages Weblate will just wait until you fix that before updating the other translations.

You need to remove one of the duplicates on your repository and once the error is gone it will continue with the other processes.

You don't need 2 English files, it is enough with the .pot. weblate will do a virtual English file.

It is better to use the .pot as source, because then you can add new languages from it.

**Reply by nijel** [reply]
Duplicate language alert should only skip one of the duplicated files and process the rest.
